lab8_9/subtask_11.py

```diff
@@ -0,0 +1,37 @@
+import re
+from utils import get_number
+
+
+def to_binary(decimal_value: int):
+    binary = ''
+    while decimal_value:
+        binary += str(decimal_value % 2)
+        decimal_value //= 2
+    return binary[::-1]
+
+
+def to_decimal(binary_value: str):
+    decimal = 0
+    power = 0
+    for i in reversed(range(len(binary_value))):
+        digit = int(binary_value[i])
+        decimal += digit * 2 ** power
+        power += 1
+    return decimal
+
+
+covert_to_binary = get_number('Enter 0 to convert decimal to binary,'
+                              'any other value to convert binary to decimal: ', int)
+
+if covert_to_binary:
+    binary_pattern = r'^[01]+$'
+    while True:
+        binary_string = input('Enter binary string: ')
+        if not re.match(binary_pattern, binary_string):
+            print('binary string can only contain 1s and 0s, try again')
+            continue
+        print(f'{binary_string} in decimal is {to_decimal(binary_string)}')
+        break
```

Comment:
would be easier if you just said user to enter a number with 'b' at the end to show this is binary.
for ex. 
"0101010101b"
"10b"